src/BlockTemplatesController.php

```diff
@@ -137,6 +163,18 @@ public function render_block_template() {
 			$this->default_block_template_is_available( 'single-product' )
 		) {
 			add_filter( 'woocommerce_has_block_template', '__return_true', 10, 0 );
+		} elseif (
```

Comment:
Correct me if I'm wrong but it looks like we're doing the same in all cases, right?

It's not a blocker but maybe we could make this a bit more readable with a single `if` and moving the checks to variables. Something along the lines of

```
if ( $show_single_product || $show_taxonomy_product_cat || $show_archive_product ) {
	add_filter( 'woocommerce_has_block_template', '__return_true', 10, 0 );
}
```

Reply:
We could potentially clean this up I agree. Something for a future PR like you suggested so we can merge this in for the other week outstanding.